### PR TITLE
Add postgres-kmeans extension formula

### PR DESCRIPTION
**Pull request by rburhum**
This module implements K-means clustering algorithm as a user-defined window function in PostgreSQL. K-means clustering is a simple way to classify data set by N-dimension vector.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Please run `brew audit --strict --online` and fix all the warnings. Thanks!

**Comment by rburhum** [comment]
Alright, I ran `brew audit --strict --online postgres-kmeans` and fixed all the warnings (as well as added a test).

Everything passes without a warning, although I feel there is an issue with the uninstall and I have been inspecting all the formulas to see how uninstalls can be overriden with no luck.

The installation creates the following files:
'/usr/local/share/postgresql/extension'/kmeans.control
'/usr/local/share/postgresql/extension'/kmeans--1.1.0.sql '/usr/local/share/postgresql/extension'/kmeans.sql
'/usr/local/lib/postgresql'/kmeans.so
'/usr/local/Cellar/postgresql/9.4.5_2/share/doc/postgresql/extension'/doc/kmeans.md

so during uninstall I need to delete these files. so brew uninstall needs to either run `make uninstall` or
run:

```bash
rm -f '/usr/local/share/postgresql/extension'/kmeans.control
rm -f '/usr/local/share/postgresql/extension'/kmeans--1.1.0.sql '/usr/local/share/postgresql/extension'/kmeans.sql
rm -f '/usr/local/lib/postgresql'/kmeans.so
rm -f '/usr/local/Cellar/postgresql/9.4.5_2/share/doc/postgresql/extension'/doc/kmeans.md
```

I realize Homebrew works with soft links, but I don't see any example of how to delete those soft links. Any ideas @mikemcquaid ?

File: Library/Formula/postgres-kmeans.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  url "http://api.pgxn.org/dist/kmeans/1.1.0/kmeans-1.1.0.zip"
+  sha256 "1fafeb16e0485eb8dfbaaff83469ea593e91844832ce5e673b79ac3f5c915225"
+
+  head "https://github.com/umitanuki/kmeans-postgresql.git"
+
+  depends_on "postgresql"
+
+  def install
+    ENV["USE_PGXS"] = "1"
+    system "make", "install"
```
Comment (mikemcquaid) [inline Library/Formula/postgres-kmeans.rb]: The `prefix` will need to be set somehow.

Comment (cosmo0920) [inline Library/Formula/postgres-kmeans.rb]: Note that [postgres-kmeans](https://github.com/umitanuki/kmeans-postgresql) does not have configure script.
This extension' s `Makefile` is written in similar way as `pgroonga`.

We can use `pgroonga` like style installation: https://github.com/Homebrew/homebrew/blob/a3f926d0b5384f320247f0555ca609d9839641be/Library/Formula/pgroonga.rb#L20-L25

@rburhum Could you try to use above style in install stage?

**Comment by DomT4** [comment]
Closing as dead. Please feel free to resubmit with comments addressed if desired. Thank for your submission to Homebrew!

**Pull request closed without merging**